### PR TITLE
eds: use batched host set updates

### PR DESCRIPTION
**Pull request by snowp**
This changes EDS to use the newly introduced batched host update. This
ensures that all host set updates are applied before triggering
membership callbacks. The benefit of this is that connection pools
will no longer be drained if a host is moved between priorities.

Signed-off-by: Snow Pettersen <snowp@squareup.com>

Risk Level: Medium
Testing: Integration test
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

File: source/common/upstream/upstream_impl.cc [hunk 1]
```diff
@@ -200,9 +200,10 @@ bool updateHealthFlag(const Host& updated_host, Host& existing_host, Host::Healt
 HostVector filterHosts(const std::unordered_set<HostSharedPtr>& hosts,
                        const std::unordered_set<HostSharedPtr>& excluded_hosts) {
   HostVector net_hosts;
+  net_hosts.reserve(hosts.size());
 
   std::set_difference(hosts.begin(), hosts.end(), excluded_hosts.begin(), excluded_hosts.end(),
-                      net_hosts.begin());
+                      std::inserter(net_hosts, net_hosts.begin()));
```
Comment (htuch) [inline source/common/upstream/upstream_impl.cc]: Why was this needed?

Comment (snowp) [reply source/common/upstream/upstream_impl.cc]: Without this tests were failing from what seemed to be insertions after the end of the vector. Guessing this was a bug that wasn't caught in the previous PR for some reason, which is strange since this code would have been exercised.
